File: flutter_app/lib/screens/order_detail_screen.dart
import 'package:flutter/material.dart';
import 'package:jeh_store_flutter/services/api_service.dart';

class OrderDetailScreen extends StatefulWidget {
  const OrderDetailScreen({super.key});

  @override
  State<OrderDetailScreen> createState() => _OrderDetailScreenState();
}

class _OrderDetailScreenState extends State<OrderDetailScreen> {
  late ApiService _apiService;
  int _orderId = 0;
  Map<String, dynamic>? _order;
  bool _isLoading = true;
  bool _isPaying = false;

  @override
  void didChangeDependencies() {
    super.didChangeDependencies();
    final args = ModalRoute.of(context)!.settings.arguments as Map;
    _apiService = args['apiService'] as ApiService;
    _orderId = args['orderId'] as int;
    _loadOrder();
  }

  Future<void> _loadOrder() async {
    setState(() => _isLoading = true);
    try {
      _order = await _apiService.getOrderDetail(_orderId);
    } catch (e) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text('Error: ${e.toString().replaceAll("Exception: ", "")}')),
        );
      }
    }
    if (mounted) setState(() => _isLoading = false);
  }

  Future<void> _pay(String method) async {
    setState(() => _isPaying = true);
    try {
      await _apiService.processPayment(_orderId, method);
      if (!mounted) return;
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(content: Text(method == 'cod' ? 'Order placed with Cash on Delivery' : 'Payment successful!')),
      );
      _loadOrder();
    } catch (e) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text('Payment failed: $e')),
        );
      }
    }
    if (mounted) setState(() => _isPaying = false);
  }

  Color _statusColor(String status) {
    switch (status) {
      case 'pending': return Colors.orange;
      case 'approved': return Colors.blue;
      case 'shipped': return Colors.indigo;
      case 'delivered': return Colors.green;
      case 'cancelled': return Colors.red;
      default: return Colors.grey;
    }
  }

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);

    return Scaffold(
      appBar: AppBar(title: Text('Order #$_orderId')),
      body: _isLoading
          ? const Center(child: CircularProgressIndicator())
          : _order == null
              ? const Center(child: Text('Order not found'))
              : SingleChildScrollView(
                  padding: const EdgeInsets.all(16),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Row(
                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
                        children: [
                          const Text('Status:', style: TextStyle(fontWeight: FontWeight.w500)),
                          Container(
                            padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 4),
                            decoration: BoxDecoration(
                              color: _statusColor(_order!['status'] as String).withOpacity(0.15),
                              borderRadius: BorderRadius.circular(12),
                            ),
                            child: Text(
                              (_order!['status'] as String).toUpperCase(),
                              style: TextStyle(
                                color: _statusColor(_order!['status'] as String),
                                fontWeight: FontWeight.bold,
                              ),
                            ),
                          ),
                        ],
                      ),
                      const SizedBox(height: 16),
                      Text(
                        'Total: \$${(_order!['total_amount'] as num).toStringAsFixed(2)}',
                        style: theme.textTheme.headlineSmall?.copyWith(
                          fontWeight: FontWeight.bold,
                          color: theme.colorScheme.primary,
                        ),
                      ),
                      if (_order!['placed_at'] != null) ...[
                        const SizedBox(height: 4),
                        Text('Placed at: ${_order!['placed_at']}',
                          style: theme.textTheme.bodySmall?.copyWith(
                            color: theme.colorScheme.onSurfaceVariant,
                          ),
                        ),
                      ],
                      const SizedBox(height: 24),

                      // Order items
                      Text('Items', style: theme.textTheme.titleLarge?.copyWith(fontWeight: FontWeight.bold)),
                      const SizedBox(height: 8),

                      if (_order!['items'] != null)
                        ...(_order!['items'] as List).map((item) => Card(
                          margin: const EdgeInsets.only(bottom: 8),
                          child: Padding(
                            padding: const EdgeInsets.all(12),
                            child: Row(
                              mainAxisAlignment: MainAxisAlignment.spaceBetween,
                              children: [
                                Expanded(
                                  child: Column(
                                    crossAxisAlignment: CrossAxisAlignment.start,
                                    children: [
                                      Text(
                                        item['title'] ?? 'Product #${item['product_id']}',
                                        style: const TextStyle(fontWeight: FontWeight.w500),
                                      ),
                                      Text(
                                        '\$${(item['unit_price'] as num).toStringAsFixed(2)} x ${item['quantity']}',
                                        style: theme.textTheme.bodySmall,
                                      ),
                                    ],
                                  ),
                                ),
                                Text(
                                  '\$${((item['unit_price'] as num).toDouble() * (item['quantity'] as int)).toStringAsFixed(2)}',
                                  style: const TextStyle(fontWeight: FontWeight.bold),
                                ),
                              ],
                            ),
                          ),
                        )),

                      const SizedBox(height: 24),

                      // Payment section for pending orders
                      if (_order!['status'] == 'pending') ...[
                        Text('Payment', style: theme.textTheme.titleLarge?.copyWith(fontWeight: FontWeight.bold)),
                        const SizedBox(height: 12),
                        SizedBox(
                          width: double.infinity,
                          height: 50,
                          child: ElevatedButton(
                            onPressed: _isPaying ? null : () => _pay('credit_card'),
                            child: _isPaying
                                ? const SizedBox(height: 24, width: 24, child: CircularProgressIndicator(strokeWidth: 2))
                                : Text('Pay Now - \$${(_order!['total_amount'] as num).toStringAsFixed(2)}',
                                    style: const TextStyle(fontSize: 16)),
                          ),
                        ),
                        const SizedBox(height: 8),
                        SizedBox(
                          width: double.infinity,
                          height: 50,
                          child: OutlinedButton(
                            onPressed: _isPaying ? null : () => _pay('cod'),
                            child: const Text('Cash on Delivery', style: TextStyle(fontSize: 16)),
                          ),
                        ),
                      ],
                    ],
                  ),
                ),
    );
  }
}

File: JEHSTORE/app/src/main/java/com/example/jeh_store/screens/OrderDetailScreen.kt
package com.example.jeh_store.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.jeh_store.models.Order
import com.example.jeh_store.services.ApiService
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun OrderDetailScreen(
    apiService: ApiService,
    orderId: Int,
    onBack: () -> Unit
) {
    var order by remember { mutableStateOf<Order?>(null) }
    var isLoading by remember { mutableStateOf(true) }
    var isPaying by remember { mutableStateOf(false) }
    var paymentMessage by remember { mutableStateOf<String?>(null) }
    val scope = rememberCoroutineScope()

    LaunchedEffect(orderId) {
        scope.launch {
            order = apiService.getOrderDetail(orderId)
            isLoading = false
        }
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Order #$orderId") },
                navigationIcon = { TextButton(onClick = onBack) { Text("Back") } }
            )
        }
    ) { padding ->
        if (isLoading) {
            Box(modifier = Modifier.fillMaxSize().padding(padding), contentAlignment = Alignment.Center) {
                CircularProgressIndicator()
            }
        } else if (order != null) {
            Column(
                modifier = Modifier.fillMaxSize().padding(padding).padding(16.dp)
            ) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween
                ) {
                    Text("Status:", fontWeight = FontWeight.Medium)
                    StatusChip(status = order!!.status)
                }

                Spacer(modifier = Modifier.height(12.dp))

                Text(
                    text = "Total: $${String.format("%.2f", order!!.total_amount)}",
                    fontWeight = FontWeight.Bold,
                    fontSize = 24.sp,
                    color = MaterialTheme.colorScheme.primary
                )

                if (order!!.placed_at != null) {
                    Text(
                        text = "Placed at: ${order!!.placed_at}",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }

                Spacer(modifier = Modifier.height(24.dp))

                Text(
                    text = "Items (${order!!.items?.size ?: 0})",
                    fontWeight = FontWeight.Bold,
                    fontSize = 18.sp
                )

                Spacer(modifier = Modifier.height(8.dp))

                order!!.items?.forEach { item ->
                    Card(
                        modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp),
                        elevation = CardDefaults.cardElevation(1.dp)
                    ) {
                        Row(
                            modifier = Modifier.padding(12.dp).fillMaxWidth(),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Column(modifier = Modifier.weight(1f)) {
                                Text(
                                    text = item.title ?: "Product #${item.product_id}",
                                    fontWeight = FontWeight.Medium
                                )
                                Text(
                                    text = "$${String.format("%.2f", item.unit_price)} x ${item.quantity}",
                                    style = MaterialTheme.typography.bodySmall
                                )
                            }
                            Text(
                                text = "$${String.format("%.2f", item.unit_price * item.quantity)}",
                                fontWeight = FontWeight.SemiBold
                            )
                        }
                    }
                }

                Spacer(modifier = Modifier.height(24.dp))

                // Payment section for pending orders
                if (order!!.status == "pending" && paymentMessage == null) {
                    Text("Payment", fontWeight = FontWeight.Bold, fontSize = 18.sp)
                    Spacer(modifier = Modifier.height(8.dp))
                    Button(
                        onClick = {
                            scope.launch {
                                isPaying = true
                                val success = apiService.processPayment(orderId, "credit_card")
                                isPaying = false
                                paymentMessage = if (success) "Payment successful!" else "Payment failed"
                                if (success) {
                                    order = apiService.getOrderDetail(orderId)
                                }
                            }
                        },
                        modifier = Modifier.fillMaxWidth().height(50.dp),
                        enabled = !isPaying
                    ) {
                        if (isPaying) CircularProgressIndicator(modifier = Modifier.size(24.dp))
                        else Text("Pay Now - $${String.format("%.2f", order!!.total_amount)}", fontSize = 16.sp)
                    }

                    OutlinedButton(
                        onClick = {
                            scope.launch {
                                isPaying = true
                                val success = apiService.processPayment(orderId, "cod")
                                isPaying = false
                                paymentMessage = if (success) "Order placed with Cash on Delivery" else "Failed"
                                if (success) {
                                    order = apiService.getOrderDetail(orderId)
                                }
                            }
                        },
                        modifier = Modifier.fillMaxWidth().height(50.dp),
                        enabled = !isPaying
                    ) {
                        Text("Cash on Delivery", fontSize = 16.sp)
                    }
                }

                if (paymentMessage != null) {
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        text = paymentMessage!!,
                        color = if (paymentMessage!!.contains("successful", ignoreCase = true) || paymentMessage!!.contains("placed", ignoreCase = true))
                            MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.error,
                        fontWeight = FontWeight.Medium
                    )
                }
            }
        } else {
            Box(modifier = Modifier.fillMaxSize().padding(padding), contentAlignment = Alignment.Center) {
                Text("Order not found")
            }
        }
    }
}